### PR TITLE
add Make targets to deploy operator

### PR DESCRIPTION
**Pull request by jcantrill**
Add Make targets to simplify deployment

### CODE REVIEW

File: hack/common [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+IMAGE_TAG=${IMAGE_TAG:-"quay.io/openshift/elasticsearch-operator:latest"}
+
+# probably needs some work for contexts which have '-' in the name and are not IPs
+context=$(oc config current-context)
+API_SERVER=$(python -c \
+    "import re; m=re.match('.*/(.*)/.*',\"${context}\"); print m.group(1).replace('-','.')")
+
+ADMIN_USER=${ADMIN_USER:-admin}
+ADMIN_PSWD=${ADMIN_USER:-admin123}
```
Comment (richm) [inline hack/common]: what are the username and password used for?

File: hack/cr.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: "elasticsearch"
 spec:
   nodeSpec:
-    image: openshift/origin-logging-elasticsearch5:latest
+    image: 172.30.1.1:5000/openshift/logging-elasticsearch5:latest
```
Comment (richm) [inline hack/cr.yaml]: why is this necessary?  It seems to work without it, in my testing

File: hack/deploy-image.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+echo "Setting up port-forwarding to remote docker-registry..."
+oc port-forward service/docker-registry -n default ${LOCAL_PORT}:${registry_port} &
+forwarding_pid=$!
+
+trap "kill -15 ${forwarding_pid}" EXIT
+sleep 1.0
+
+echo "Pushing image ${tag}..."
+docker login 127.0.0.1:${LOCAL_PORT} -u ${ADMIN_USER} -p $(oc whoami -t)
```
Comment (richm) [inline hack/deploy-image.sh]: where is the `oc login` for the user and password?

In my testing, I did not use `deploy-image` at all and tests seemed to work fine.

File: hack/deploy-setup.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+load_manifest() {
+  local repo=$1
+  local namespace=${2:-}
+  if [ -n "${namespace}" ] ; then
+    namespace="-n ${namespace}"
+  fi
+
+  pushd ${repo}/manifests
+    for m in $(ls); do
+      if [ "$(echo ${EXCLUSIONS[@]} | grep -o ${m} | wc -w)" == "0" ] ; then
```
Comment (richm) [inline hack/deploy-setup.sh]: I guess this works - but usually if you use the old style bourne shell syntax `if [` you have to use `=` and if you use the new style bash `if [[` you have to use `==`

File: hack/deploy-setup.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        oc create -f ${m} ${namespace:-} ||:
```
Comment (richm) [inline hack/deploy-setup.sh]: are you ignoring the failure here because the objects may already exist?
ignoring errors here will mask many errors, not just `already exists`

File: hack/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -e
+
+source "$(dirname $0)/common"
+
+registry_ip=$(oc get service docker-registry -n default -o jsonpath={.spec.clusterIP})
+cat manifests/05-deployment.yaml | \
+    sed -e "s,${IMAGE_TAG},${registry_ip}:5000/openshift/elasticsearch-operator:latest," | \
+	oc create -n ${NAMESPACE} -f -
```
Comment (richm) [inline hack/deploy.sh]: why is this necessary?  It seemed to work for me without this.

File: hack/cr.yaml (hunk 2 above)
Comment (jcantrill) [reply hack/cr.yaml]: Assuming we build and push the image with this to dockerhub this change can be omitted.  I will investigate

Comment (richm) [reply hack/cr.yaml]: a test deploy or test-e2e should not push an image to an external registry
are you saying that if you build a local image, you must push it to the local registry in order for it to be used?  For example
```
cd origin-aggregated-logging
docker build -t openshift/origin-logging-elasticsearch5:latest -f elasticsearch/Dockerfile.centos7 elasticsearch/
```
This creates a local image `openshift/origin-logging-elasticsearch5:latest` - `docker images` shows this image

Are you saying that if I then deploy elasticsearch, it will attempt to pull `openshift/origin-logging-elasticsearch5:latest` from `hub.docker.io` or `quay.io`, even though the image is available locally?

File: hack/deploy-setup.sh [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  done
+popd
+
+oc create -f hack/prometheus-operator-crd-cluster-roles.yaml ||:
+
+oc create clusterrolebinding elasticsearch-operator-prometheus-rolebinding \
+    --serviceaccount=${NAMESPACE}:elasticsearch-operator \
+    --clusterrole=prometheus-crd-edit ||:
+
+# This is necessary for running the operator with go run
```
Comment (ewolinetz) [inline hack/deploy-setup.sh]: This should only be necessary for `Cluster Logging Operator`. Elasticsearch Operator doesn't leverage a working dir does it?

File: hack/prometheus-operator-crd-cluster-roles.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-crd-view
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+- apiGroups: ["monitoring.coreos.com"]
```
Comment (ewolinetz) [inline hack/prometheus-operator-crd-cluster-roles.yaml]: this doesn't match the roles in the manifest file. Do we need this since we should be using the manifest files?
